udp/connection.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2015 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package udp
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+)
+
+type ConnectionIDGenerator struct {
+	iv, iv2 []byte
+	block   cipher.Block
+}
+
+func (g *ConnectionIDGenerator) Init() error {
```

Comment:
Can we get a constructor for ConnectionIDGenerator? This could become just that.